gapic/src/main/com/google/gapid/widgets/LinkifiedTree.java

```diff
@@ -411,9 +420,25 @@ private boolean shouldIgnoreColors(Event event) {
       return (event.detail & SWT.SELECTED) != 0;
     }
 
+    public void updateHierarchy(TreeItem item) {
+      while(item != null) {
+        updateWithChildren(item);
+        item = item.getParentItem();
+      }
+    }
+
+    private void updateWithChildren(TreeItem item) {
+      update(item);
+      Arrays.asList(item.getItems()).forEach(child -> updateWithChildren(child));
+    }
+
     private void update(TreeItem item) {
-      Label label = getLabelNoUpdate(item);
       T element = getElement(item);
+      if (element == null) {
+        return;
+      }
```

Comment:
This seems fishy. Why do we now need this guard?

Reply:
When we traverse the tree, we get some artificial nodes without data. It's to prevent that.